pubtools/_quay/push_docker.py

```diff
@@ -731,7 +730,17 @@ def run(self) -> None:
                 failed = True
 
         # Remove old signatures
-        outdated_manifests = self.get_outdated_manifests(backup_tags)
+        # run generate backup mapping again tu push new digest of pushed containers
+        backup_tags2, _ = self.generate_backup_mapping(docker_push_items)
+        # if new backup tag has differnet digest, it means it was overwritten during the push
+        # and old signature should be removed. If manifest is the same it means, same item
+        # was just repushed
+        outdated_tags = {}
+        for bt1, bt2 in zip(backup_tags.items(), backup_tags2.items()):
+            if bt1[1] != bt2[1]:
```

Comment:
Is it safe to assume that the data from the two calls would correspond correctly? Wouldn't the second call of generate_backup_mapping return more backup tags - those that were previously rollback tags?

Reply:
yeah, that's why backup_tags2_shared is used instead of backup_tags2